src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -545,27 +545,27 @@ class _DataHookSelector:
         default=("on_before_batch_transfer", "transfer_batch_to_device", "on_after_batch_transfer")
     )
 
-    def get_hook(self, hook_name: str) -> Callable:
+    def get_instance(self, hook_name: str) -> Callable:
```

Comment:
@awaelchli might've chosen hook for a specific reason

Reply:
I added this.

Reply:
Nonetheless, the dataclass loses it's meaning. 
Now, the responsibility of mapping the callable is no longer here. You need to update the design and description.

Reply:
The return type needs to be  updated too.

Reply:
updated!